src/services/types.ts

```diff
@@ -575,8 +575,16 @@ namespace ts {
 
     export interface CombinedCodeFixScope { type: "file"; fileName: string; }
 
+    export const enum OrganizeImportsMode {
+        All = "All",
+        SortAndCombine = "SortAndCombine",
+        RemoveUnused = "RemoveUnused",
+    }

```

Comment:
I initially had `SortImports`, `SortSpecifiers`, `CoalesceImports`, and `RemoveUnused` as bit flags, but it turned out to be a lot of effort to separate sorting and combining, which nobody was asking for. If we do need to add more modes in the future (unlikely), their names could get ugly, but named modes allow us to invest relatively little now while giving ourselves a little room to grow.

Reply:
I like these options..

Reply:
The implementation looks like it really, really wants to consume a flags enum.  What if we just merged the flags that aren't manipulated independently?

Reply:
Because if we ever decide to break them down more granularly, we can’t without a breaking API change.